MatrixSDKTests/MXBaseKeyBackupTests.m

```diff
@@ -509,26 +509,28 @@ - (void)testCrossSigningMSKTrustForKeyBackupVersion
             // - Alice creates a backup
             [aliceSession.crypto.backup prepareKeyBackupVersionWithPassword:nil algorithm:self.algorithm success:^(MXMegolmBackupCreationInfo *keyBackupCreationInfo) {
                 [aliceSession.crypto.backup createKeyBackupVersion:keyBackupCreationInfo success:^(MXKeyBackupVersion *keyBackupVersion) {
-
+                    // De-serialize the authData, find the MSK signature, remove all other signatures from the authData, serialize it back in
+                    MXMegolmBackupAuthData *authData = [MXMegolmBackupAuthData modelFromJSON:keyBackupVersion.authData];
+                    NSString *myMSKSignatureKey = [NSString stringWithFormat:@"ed25519:%@", aliceSession.crypto.crossSigning.myUserCrossSigningKeys.masterKeys.keys];
+                    authData.signatures = @{
+                        aliceSession.myUserId: @{
+                            myMSKSignatureKey: authData.signatures[aliceSession.myUserId][myMSKSignatureKey]
+                        }
+                    };
+                    keyBackupVersion.authData = [authData JSONDictionary];
+                    
                     // - Check the returned MXKeyBackupVersion is trusted
                     [aliceSession.crypto.backup trustForKeyBackupVersion:keyBackupVersion onComplete:^(MXKeyBackupVersionTrust *keyBackupVersionTrust) {
-
-                        // -> It must be trusted by 2 entities
+                        // -> It must be trusted by 1 entity
                         XCTAssertNotNil(keyBackupVersionTrust);
                         XCTAssertTrue(keyBackupVersionTrust.usable);
-                        XCTAssertEqual(keyBackupVersionTrust.signatures.count, 2);
```

Comment:
The new test looks good, however it might be better to create it as copy and leave testCrossSigningMSKTrustForKeyBackupVersion as is, otherwise we loose the test case for getting a device signature (XCTAssertEqual(keyBackupVersionTrust.signatures.count, 2))